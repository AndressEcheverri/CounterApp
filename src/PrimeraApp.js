//import React, { Fragment } from 'react';
import React from 'react';
import { PropTypes } from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {

    return(
        <>

            <h1>{ saludo }</h1>
            <p>{ subtitulo }</p>

        </>
       
    );

}
 PrimeraApp.propTypes = {
     saludo: PropTypes.string.isRequired //esta es la manera de obligar a los otros programadores de usar mi componente como lo digo
 }

PrimeraApp.defaultProps = {

    subtitulo:'Soy un subtitulo'
}

export default PrimeraApp;