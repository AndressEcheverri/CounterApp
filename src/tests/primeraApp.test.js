import React from 'react';
import '@testing-library/jest-dom';
//import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

import { shallow } from 'enzyme';

describe('Pruebas en primeraApp.js', () => {

   // test('debe mostrar el mensaje "hola, soy pipe" ', () => {
        
     //   const saludo = 'Hola,soy pipe';

       // const { getByText } = render( <PrimeraApp saludo={ saludo } />);

       // expect( getByText(saludo) ).toBeInTheDocument();

    //})

    test('debe mostrar <PrimeraApp /> correctamente', () => {

        const saludo = 'hola, soy pipe';
        const wrapper = shallow(<PrimeraApp saludo = { saludo } />);

        expect( wrapper ).toMatchSnapshot();
        
    });


    test('debe mostrar el subtitulo enviado por props', () => {

        const saludo = 'hola, soy pipe';
        const subtitulo = 'hola, soy un subtitulo';

        const wrapper = shallow(
                <PrimeraApp 
                
                    saludo = { saludo }
                    subtitulo = { subtitulo }
                
                />
        );


        const textoParrafo = wrapper.find('p').text();

        expect( textoParrafo ).toBe( subtitulo );
            
    });

   
    
    
    
});
