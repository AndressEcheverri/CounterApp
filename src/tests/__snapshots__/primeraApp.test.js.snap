// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primeraApp.js debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hola, soy pipe
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
